conan/tools/gnu/pkgconfig.py

```diff
@@ -23,11 +24,16 @@ def __init__(self, conanfile, library, pkg_config_path=None):
     def _parse_output(self, option):
         executable = self._conanfile.conf.get("tools.gnu:pkg_config", default="pkg-config")
         command = cmd_args_to_string([executable, '--' + option, self._library, '--print-errors'])
+
         env = Environment()
         if self._pkg_config_path:
             env.prepend_path("PKG_CONFIG_PATH", self._pkg_config_path)
         with env.vars(self._conanfile).apply():
-            return check_output_runner(command).strip()
+            # This way we get the environment from ConanFile, from profile (default buildenv)
+            output = StringIO()
+            self._conanfile.run(command, output)
```

Comment:
Let's use the kwarg to avoid confusion:

 ```suggestion
             self._conanfile.run(command, stdout=output)
```